luarules/gadgets/unit_gunship_strafe_fix.lua

```diff
@@ -0,0 +1,100 @@
+local gadget = gadget ---@type Gadget
+
+function gadget:GetInfo()
+	return {
+		name = "Gunship Strafe Fix",
+		desc = "When gunships stop attacking their target unit with no followup command, forces them to stop strafing and enter idle behavior",
+		author = "JRTaylord: https://github.com/JRTaylord",
+		date = "May 7, 2025",
+		license = "GNU GPL, v2 or later",
+		layer = 0, -- Todo: ask which layer should this be on
+		enabled = true,
+	}
+end
+
+local spGiveOrderToUnit = Spring.GiveOrderToUnit
+local spGetUnitWeaponTarget = Spring.GetUnitWeaponTarget
+local spGetUnitCommands = Spring.GetUnitCommands
+local spGetAllUnits = Spring.GetAllUnits
+local spGetUnitDefID = Spring.GetUnitDefID
+local spEcho = Spring.Echo
+
+-- Constants
+-- Todo figure out options to make the command silent
+local CMD_OPTIONS = CMD.OPT_INTERNAL
+
+-- Gadget variables
+local gunshipDefs = {}
+local gunshipsToTrack = {}
+
+-- Only run in synced code
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
```

Comment:
Should probably put this higher, adjacent to GetInfo, to avoid doing needless work.

Reply:
Will do